substrate/client/rpc-spec-v2/src/transaction/metrics.rs

```diff
@@ -0,0 +1,184 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Metrics for recording transaction events.
+
+use std::time::Instant;
+
+use prometheus_endpoint::{
+	register, CounterVec, HistogramOpts, HistogramVec, Opts, PrometheusError, Registry, U64,
+};
+
+use super::TransactionEvent;
+
+/// Histogram time buckets in microseconds.
+const HISTOGRAM_BUCKETS: [f64; 11] = [
+	5.0,
+	25.0,
+	100.0,
+	500.0,
+	1_000.0,
+	2_500.0,
+	10_000.0,
+	25_000.0,
+	100_000.0,
+	1_000_000.0,
+	10_000_000.0,
+];
+
+/// Labels for transaction status.
+mod labels {
```

Comment:
nit: I am thinking enums enums + strum's [IntoStaticStr](https://docs.rs/strum/latest/strum/derive.IntoStaticStr.html) deriving can work too here, which improves type safety. Is there a strong reason to prefer static strs over enums, other than simplicity?

Reply:
Yep that makes sense, I ll have a look at integrating that here, thanks for the suggestion 🙏

Reply:
Decided not to go on this path anymore as we'll need to match the transaction event anyway to get the respective `Histogram` (and there's also the `BestChainBlockIncluded(Some(_))` and `BestChainBlockIncluded(None)` differences in events reported)